romeo/asset-contract/tests/asset_SetBitcoinWalletPublicKeyCommand_NonOwner.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  AssetCommand,
+  Real,
+  Stub,
+  shortenString
+} from "./asset_CommandModel.ts";
+
+import { tx } from "@hirosystems/clarinet-sdk";
+import { Cl } from "@stacks/transactions";
+
+import { expect } from "vitest";
+
+export class SetBitcoinWalletPublicKeyCommand_NonOwner implements AssetCommand {
+  readonly sender: string;
+  readonly pubKey: Uint8Array;
+
+  constructor(
+    sender: string,
+    pubKey: Uint8Array,
+  ) {
+    this.sender = sender;
+    this.pubKey = pubKey;
+  }
+
+  check(_: Readonly<Stub>): boolean {
+    // Can run if sender is not the deployer. This is filtered in the generator.
+    return true;
```

Comment:
Logic of generator should be enforced here for better understanding.

Reply:
I disagree. This would skew command selection, as explained by Moodmosaic. Ambiguity in input generation should be resolved 'at' the generators.

Reply:
See https://github.com/stacks-network/sbtc/pull/352#discussion_r1385614571. But in reality, what @friedger writes makes sense to me; it'd be much easier if everything is in one place.

Perhaps when we switch to [defining commands on-the-fly](https://github.com/dubzzz/fast-check/discussions/3026) things will be more clear, since both the generator and the command will be in the 'same' place.